src/Microsoft.IdentityModel.Tokens/EventBasedLRUCache.cs

```diff
@@ -50,34 +50,22 @@ internal class EventBasedLRUCache<TKey, TValue>
         // if true, expired values will not be added to the cache and clean-up of expired values will occur on a 5 minute interval
         private readonly bool _removeExpiredValues;
         private readonly int _removeExpiredValuesIntervalInSeconds;
-        // for testing purpose only to verify the task count
-        private int _taskCount;
         private DateTime _timeForNextExpiredValuesRemoval;
-        private DateTime _timeForNextCompaction;
 
         #region event queue
-        private int _eventQueuePollingInterval = 50;
-        // The idle timeout, the _eventQueueTask will end after being idle for the specified time interval (execution continues even if the queue is empty to reduce the task startup overhead), default to 120 seconds.
-        // TODO: consider implementing a better algorithm that tracks and predicts the usage patterns and adjusts this value dynamically.
-        private long _eventQueueTaskIdleTimeoutInSeconds = 120;
-        // The time when the _eventQueueTask should end. The intent is to reduce the overhead costs of starting/ending tasks too frequently
-        // but at the same time keep the _eventQueueTask a short running task.
-        // Since Task is based on thread pool the overhead should be reasonable.
-        private DateTime _eventQueueTaskStopTime;
-        // task states used to ensure thread safety (Interlocked.CompareExchange)
-        private const int EventQueueTaskStopped = 0; // task not started yet
-        private const int EventQueueTaskRunning = 1; // task is running
-        private const int EventQueueTaskDoNotStop = 2; // force the task to continue even it has past the _eventQueueTaskStopTime, see StartEventQueueTaskIfNotRunning() for more details.
-        private int _eventQueueTaskState = EventQueueTaskStopped;
-
         private const int ActionNotQueued = 0; // compaction action not in the event queue
         private const int ActionQueuedOrRunning = 1; // compaction action in the event queue or currently in progress
+
         private int _compactValuesState = ActionNotQueued;
         private int _removeExpiredValuesState = ActionNotQueued;
         private int _processCompactedValuesState = ActionNotQueued;
+        private readonly Task _eventQTask;
+        private int _eventQueuePollingInterval = 50; // in milliseconds
+        // for testing purpose only to verify the task count
+        private int _taskCount;
```

Comment:
this doesn't look relevant anymore.

Reply:
I agree, I will set it to 1.
There is an internal TaskCount method that we should keep.